packages/react/src/components/Tile/Tile.tsx

```diff
@@ -497,48 +502,41 @@ export const SelectableTile = React.forwardRef<
     className
   );
 
-  function handleClick(evt) {
+  // Single function to handle selection changes
+  const handleSelectionChange = useCallback(
+    (evt: React.SyntheticEvent, newSelected: boolean) => {
+      setIsSelected(newSelected);
+      onChange(evt as React.ChangeEvent<HTMLDivElement>, newSelected, id);
+    },
+    [onChange, id]
+  );
+
+  function handleClick(evt: React.MouseEvent<HTMLDivElement>) {
     evt.preventDefault();
     evt?.persist?.();
     if (
       normalizedDecorator &&
       decoratorRef.current &&
-      decoratorRef.current.contains(evt.target)
+      decoratorRef.current.contains(evt.target as Node)
```

Comment:
The type assertion isn't necessary. There's a type-safe way to implement this condition:

```suggestion
      evt.target instanceof Node &&
      decoratorRef.current.contains(evt.target)
```

Reply:
Have added these changes